zebra-state/src/service/finalized_state.rs

```diff
@@ -380,8 +385,20 @@ impl FinalizedState {
     }
 
     /// Returns the history tree for the finalized state.
-    pub fn history_tree(&self) -> &HistoryTree {
-        todo!("add history tree to finalized state");
+    /// TODO: this a dummy implementation to prevent zebra from panicking.
+    /// It will be properly implemented in https://github.com/ZcashFoundation/zebra/issues/2134
+    pub fn history_tree(&mut self) -> &HistoryTree {
+        // TODO: do we need to handle an empty tip?
+        let (height, _) = self.tip().expect("tip exists");
+        let tip = self
+            .block(height.into())
+            .expect("block will exist if tip does");
+        // TODO: pass roots
+        // TODO: return error?
+        let tree = HistoryTree::from_block(self.network, tip, &Default::default(), None)
```

Comment:
This history tree gets passed to `check::block_commitment_is_valid_for_chain_history`, which will fail because the history tree doesn't have the right blocks.

So we'll also need to disable `check::block_commitment_is_valid_for_chain_history`.

And at that point we might as well revert the PR. 😢

Reply:
Yep :cry: I'll close this PR since it doesn't make sense anymore.